packages/http-client-java/emitter/src/validate.ts

```diff
@@ -0,0 +1,71 @@
+import { Program } from "@typespec/compiler";
+import { logError, spawnAsync } from "./utils.js";
+
+export const JDK_NOT_FOUND_MESSAGE =
+  "Java Development Kit (JDK) is not found in PATH. Please install JDK 17 or above. Microsoft Build of OpenJDK can be downloaded from https://learn.microsoft.com/java/openjdk/download";
+
+export async function validateDependencies(
+  program: Program | undefined,
+  logDiagnostic: boolean = false,
+) {
+  // Check JDK and version
+  try {
+    const result = await spawnAsync("javac", ["-version"], { stdio: "pipe" });
```

Comment:
should we do the same win32 check as we does for `mvn`?

Reply:
`javac` is a normal exe (unlike mvn.cmd is windows script, while mvn is linux script)